### ISSUE TITLE
StackLevel too deep

### ISSUE THREAD

**Issue by jclay** [checkr/checkr-ruby#5]
Getting a stack level too deep error when creating a report:

``` ruby
Checkr::Report.create({package: "tasker_pro", candidate_id: "cd663309295a003aa5c5e59f"})  

(pry)> output error: #<SystemStackError: stack level too deep>
```

Still doing some digging to see if I can find out where this is going wrong.

**Comment by jclay** [comment]
This appears to be related to `#inspect` method on `api_class.rb`. 

Commenting it out solves the issue
```ruby
def inspect
  puts "inspect"
  # id_string = (self.respond_to?(:id) && !self.id.nil?) ? " id=#{self.id}" : ""
  # "#<#{self.class}:0x#{self.object_id.to_s(16)}#{id_string}> JSON: " + JSON.pretty_generate(attributes)
end
```

It appears to be specific to how pry automatically calls pretty print on a returned object. It looks like pretty print then calls `#inspect`

```ruby
def inspect
  puts caller[0..100].join("\n")
  # id_string = (self.respond_to?(:id) && !self.id.nil?) ? " id=#{self.id}" : ""
  # "#<#{self.class}:0x#{self.object_id.to_s(16)}#{id_string}> JSON: " + JSON.pretty_generate(attributes)
end
```

Call trace for above

```
/user_path/lib/ruby/2.2.0/pp.rb:314:in `pretty_print'
/user_path/lib/ruby/2.2.0/pp.rb:187:in `block in pp'
/user_path/lib/ruby/2.2.0/prettyprint.rb:251:in `block (2 levels) in group'
/user_path/lib/ruby/2.2.0/prettyprint.rb:278:in `nest'
/user_path/lib/ruby/2.2.0/prettyprint.rb:250:in `block in group'
/user_path/lib/ruby/2.2.0/prettyprint.rb:263:in `group_sub'
/user_path/lib/ruby/2.2.0/prettyprint.rb:249:in `group'
/user_path/lib/ruby/2.2.0/pp.rb:187:in `pp'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/color_printer.rb:34:in `pp'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/color_printer.rb:17:in `block in pp'
/user_path/lib/ruby/2.2.0/pp.rb:146:in `guard_inspect_key'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/color_printer.rb:17:in `pp'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry.rb:23:in `block (2 levels) in <class:Pry>'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/pager.rb:33:in `open'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry.rb:21:in `block in <class:Pry>'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/pry_instance.rb:367:in `call'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/pry_instance.rb:367:in `show_result'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/pry_instance.rb:336:in `block in handle_line'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/pry_class.rb:369:in `critical_section'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/pry_instance.rb:335:in `handle_line'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/pry_instance.rb:243:in `block (2 levels) in eval'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/pry_instance.rb:242:in `catch'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/pry_instance.rb:242:in `block in eval'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/pry_instance.rb:241:in `catch'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/pry_instance.rb:241:in `eval'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/repl.rb:77:in `block in repl'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/repl.rb:67:in `loop'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/repl.rb:67:in `repl'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/repl.rb:38:in `block in start'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/input_lock.rb:61:in `call'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/input_lock.rb:61:in `__with_ownership'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/input_lock.rb:79:in `with_ownership'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/repl.rb:38:in `start'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/repl.rb:15:in `start'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/pry_class.rb:169:in `start'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/cli.rb:219:in `block in <top (required)>'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/cli.rb:83:in `call'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/cli.rb:83:in `block in parse_options'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/cli.rb:83:in `each'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/lib/pry/cli.rb:83:in `parse_options'
/user_path/lib/ruby/gems/2.2.0/gems/pry-0.10.1/bin/pry:16:in `<top (required)>'
/user_path/bin/pry:23:in `load'
/user_path/bin/pry:23:in `<main>'
```

**Comment by jclay** [comment]
@joncalhoun I've been able to narrow this down to a test case for you.

This reproduces it for me consistently. 

For anyone else who comes across this - it appears to be fixed by upgrading activesupport to 4.2.4 by bumping rails to 4.2.4 in our `Gemfile`

**Comment by joncalhoun** [comment]
Is it only reproducable with pry?

**Comment by jclay** [comment]
No - I'm able to reproduce in IRB as well. The test script I attached *should* reproduce on any environment.

**Comment by jclay** [comment]
It's seems to be related how active_support 4.2.3 is overriding `to_json`.

**Comment by joncalhoun** [comment]
Not at a good place to debug at the moment, but can you try removing this part of the inspect output: `#{self.object_id.to_s(16)}`

**Comment by jclay** [comment]
No dice. 

Commenting out any calls to JSON does fix the issue.

```
    def to_s(*args)
      #JSON.pretty_generate(non_nil_attributes)
    end
```

**Comment by jkogara** [comment]
Any word on a fix for this coming soon?

**Comment by jclay** [comment]
@jkogara I fixed this by bumping rails up to `4.2.4`
